fastcrypto/src/groups/mod.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+pub mod ristretto255;
+
+pub trait AdditiveGroupElement: PartialEq + Eq + Serialize + DeserializeOwned {
+    type Group: AdditiveGroup<Element = Self>;
+}
+
+/// Trait impl'd by additive groups.
+pub trait AdditiveGroup {
```

Comment:
This may just be a perception, but the usual way in which this is addressed in a whole bunch of rust libraries is to implement `core::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign};` which `auto_ops` doesn't seem to (or does it?). 

See this example in the original Ristretto impl:
https://github.com/dalek-cryptography/curve25519-dalek/blob/90292dccf84cdc614c1a5640f2d6cb13bfff9761/src/ristretto.rs#L798-L804

We're newtyping this for encapsulation inside the fastcrypto library. I wonder if we could dramatically reduce the amount of new code we have to maintain by using those traits, specifically, the ability of the `derive_more` crate to derive them on a newtype (when the inner type satisfies them):
https://jeltef.github.io/derive_more/derive_more/#operators

The change that we'd see in this trait then is to remove the arithmetic operations, and put the definition of the `core::ops` as required trait bounds. The present trait would then only contain the stuff that isn't already covered, such as the identity.

Does that make sense?